module/zfs/dsl_crypt.c

```diff
@@ -2430,9 +2406,7 @@ dsl_crypto_populate_key_nvlist(dsl_dataset_t *ds, uint64_t from_ivset_guid,
 	VERIFY0(dmu_objset_from_ds(ds, &os));
 	mdn = DMU_META_DNODE(os);
 
-	ret = nvlist_alloc(&nvl, NV_UNIQUE_NAME, KM_SLEEP);
-	if (ret != 0)
-		goto error;
+	nvlist_alloc(&nvl, NV_UNIQUE_NAME, KM_SLEEP);
```

Comment:
nit: `fnvlist_alloc()` was added for this, can you change this to `nvl = fnvlist_alloc();`.  It's equivalent, internally it calls `nvlist_alloc()` and VERIFYs that it never fails.